### DISCUSSION TITLE
Setting UserDisplayOptions

### DISCUSSION THREAD

**Discussion by ajcaterino** [pnp/pnpcore#1260] [Q&A] [answered]
We are currently using PnP Powershell to set UserDisplayOptions for a field:

`
Set-PnPField -List $ListName -Identity "Preparer" -Values @{ 'UserDisplayOptions' = 'NamePhoto' }
`

Is there a way to accomplish this in the PnP Core SDK? I can find the UserDisplayOptions property in /src/generated/SP/Internal/FieldUser.cs class and its corresponding interface, IFieldUser.cs, but I cannot find it in the SDK source code.

**Comment by jansenbe** [comment]
@ajcaterino : did you check if that property is the field SchemaXml properyt? If so, you can update the field SchemaXml and save it.

**Reply by ajcaterino** [reply]
I'm not familiar with what you have suggested. Can you please provide some links for me to reference?

**Reply by jansenbe** [reply, marked as answer]
@ajcaterino : load the list field you want to update, read the SchemaXml property of that field (https://pnp.github.io/pnpcore/api/PnP.Core.Model.SharePoint.IField.html#PnP_Core_Model_SharePoint_IField_SchemaXml) and add/update the `UserDisplayOptions="NamePhoto"` element to the XML and then save it back (see https://pnp.github.io/pnpcore/using-the-sdk/fields-intro.html#updating-site-and-list-fields). The resulting field SchemaXml can look like this:

```xml
<Field ClientSideComponentId="00000000-0000-0000-0000-000000000000" DisplayName="PersonSingle" Format="Dropdown" List="UserInfo" Name="PersonSingle" 
Title="PersonSingle" Type="User" UserSelectionMode="1" UserSelectionScope="0" UserDisplayOptions="NamePhoto" ID="{5c2ce6f6-21f8-4399-b1a3-7c6142f720eb}" Version="2" StaticName="PersonSingle" SourceID="{7d644d41-d86e-4594-99de-479594a68fd9}" ColName="int1" RowOrdinal="0" />
```

Note that I did not try this yet myself, but it's a common approach to configure special field props.
